test/e2e/specs/editor/various/list-view.spec.js

```diff
@@ -779,6 +779,100 @@ test.describe( 'List View', () => {
 				{ name: 'core/heading', selected: false, focused: false },
 				{ name: 'core/file', selected: false, focused: true },
 			] );
+
+		// Copy and paste blocks. To begin, add another Group block.
+		await editor.insertBlock( {
+			name: 'core/group',
+			innerBlocks: [
+				{ name: 'core/paragraph' },
+				{ name: 'core/pullquote' },
+			],
+		} );
+
+		// Click the newly inserted Group block List View item to ensure it is focused.
+		await listView
+			.getByRole( 'link', {
+				name: 'Group',
+				expanded: false,
+			} )
+			.click();
+
+		// Move down to group block, expand, and then move to the paragraph block.
+		await page.keyboard.press( 'ArrowDown' );
+		await page.keyboard.press( 'ArrowRight' );
+		await page.keyboard.press( 'ArrowDown' );
+		await page.keyboard.press( 'ArrowDown' );
+		await pageUtils.pressKeys( 'primary+c' );
+		await page.keyboard.press( 'ArrowUp' );
+		await pageUtils.pressKeys( 'primary+v' );
+
+		await expect
+			.poll(
```

Comment:
That's a great question! I was just following the examples from the other tests, which use `poll`. In this case we're polling for an expected object that is returned by a function call, rather than for a particular selector. Internally, `getBlocksWithA11yAttributes` is calling `getByRole` etc to grab the row element itself, so I imagine this was an abstraction so that we can just focus on the shape of the block data, rather than selectors within these particular tests?

It looks like the abstraction was added in https://github.com/WordPress/gutenberg/pull/50422/ by @kevin940726. To my eyes it improves readability quite a bit, but Kai might have more context to offer 🙂

Reply:
I don't think we need it here, but it's just a preference of mine to make the assertion retryable to reduce potential flakiness.

It might be a common misconception that Playwright automatically waits for the elements when you query them, but it doesn't. It only waits for the element when you perform some "actions" on them (like clicking, or asserting). In `getBlocksWithA11yAttributes` we're not performing any actions so they might be outdated.

Reply:
Great, thanks for confirming the reasoning, Kai!